File: application/src/Caravelo/Modules/Survey/Domain/Service/SurveyService.php
<?php

namespace Caravelo\Modules\Survey\Domain\Service;

use Caravelo\Modules\Survey\Domain\Repository\SurveyRepositoryInterface;

class SurveyService
{
    /**
     * @var SurveyRepositoryInterface
     */
    private $surveyRepositoryInterface;

    public function __construct(SurveyRepositoryInterface $surveyRepositoryInterface)
    {
        $this->surveyRepositoryInterface = $surveyRepositoryInterface;
    }

    /**
     * @return \Caravelo\Modules\Survey\Domain\Model\Survey[]
     */
    public function getSurveys()
    {

        return $this->surveyRepositoryInterface->findAll();
    }

    /**
     * @param $slug
     * @return \Caravelo\Modules\Survey\Domain\Model\Survey | false
     */
    public function getSurvey($slug)
    {
        return $this->surveyRepositoryInterface->findOneBySlug($slug);
    }

}

File: application-java/src/main/java/survey/domain/service/SurveyService.java
package survey.domain.service;

import survey.domain.model.Survey;
import survey.domain.repository.SurveyRepositoryInterface;

public class SurveyService {
	
	private SurveyRepositoryInterface surveyRepositoryInterface;
	
	public SurveyService(SurveyRepositoryInterface surveyRepositoryInterface){
		this.surveyRepositoryInterface = surveyRepositoryInterface;
	}
	
	public Survey[] getSurveys(){
		return this.surveyRepositoryInterface.findAll();
	}
	
	public Survey getSurvey(String slug){
		return this.surveyRepositoryInterface.findOneBySlug(slug);
	}
	
}
